components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -5523,7 +5537,7 @@ def members():
                             rettype = "ErrorResult"
                         yield name, attribute_arguments(typeNeedsCx(m.type, False), m.type), rettype
 
-            if descriptor.proxy:
+            if descriptor.proxy or descriptor.hasNamedPropertiesObject():
```

Comment:
I'm on the fence about this, because the SupportedPropertyNames and NamedGetter methods won't actually be hooked up to anything automatically. Is the idea that the named properties proxy object would grab the window object and invoke those methods? I think it would be less confusing if we did not reuse this machinery in a way that's different than how it's used for proxy objects.